### PR TITLE
Docs: update links in developer guide to frontend source files

### PR DESCRIPTION
**Pull request by yhtMinceraft1010X**
This PR aims to update the links and diagrams in the DG page for HTML report to reflect new file names and locations. 

Commit Message:
```
After the transition to Vue, there are links and diagrams in the
Developer Guide documentation pages that still refer to files that
previously had .js extension and/or were located in now-deleted
folders.

Let's update these links and diagrams to reflect the new file
extensions and the updated location of these files.
```

### CODE REVIEW

File: docs/dg/report.md [hunk 1]
```diff
@@ -127,6 +127,6 @@ For ramps between the date ranges, the slices will be selected and it will be pr
 
 <!-- ==================================================================================================== -->
 
-## Segment view ([v_segment.js](https://github.com/reposense/RepoSense/blob/master/frontend/src/static/js/v_segment.js))
+## Segment view ([v_segment.vue](https://https://github.com/reposense/RepoSense/blob/master/frontend/src/components/v-segment.vue))
```
Comment (HCY123902) [inline docs/dg/report.md:130]: I think there is an additional http here

Comment (yhtMinceraft1010X) [reply docs/dg/report.md]: Fixed in a subsequent commit.

**Review by HCY123902 — APPROVED** [review]

**Comment by gerhean** [comment]
The original `main.js` is actually separated into the new `main.js` and `app.vue`, but I'm not very sure if  `app.vue` needs to be mentioned.

**Review by dcshzj — CHANGES_REQUESTED** [review]
The "Learning the Basics" page needs to be updated as well, to point the links to the new files.

**Review by dcshzj — APPROVED** [review]
LGTM!

**Review by fzdy1914 — APPROVED** [review]
Nice Work!

**Pull request merged**